### PR TITLE
Remove serum vial

### PR DESCRIPTION
**Pull request by tomjohn1028**
Closes https://github.com/mithraiclabs/psyoptions-market-tracker/issues/18

see console [here](https://api.psyoptions.io/console)

Example query to get all events by an address. Copy and paste this into the console
````
query MyQuery {
  serum_events(order_by: {timestamp: desc}, where: {open_order_account: {owner: {_eq: "StQPVao4aR5b2VJ71bB9Gb6aKnHi4kK6oDK7JVLnMRh"}}}) {
    open_order_account {
      market
      owner
      address
    }
    type
    timestamp
    slot
    size
    side
    price
    order_id
    maker
    fee_tier
    fee_cost
    client_id
    account_slot
    account
  }
}
````

### CODE REVIEW

File: listener/src/graphQLClient.ts [hunk 1]
```diff
@@ -6,11 +6,12 @@ import { execute } from 'apollo-link';
 import { WebSocketLink } from 'apollo-link-ws';
```
Comment (Taylor123) [inline listener/src/graphQLClient.ts:6]: such a nit pick, but feel like we should split this file

File: listener/src/serumListener.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      price: _getFillPrice(market, event),
+      size: _getFillSize(market, event),
+      maker: event.eventFlags.maker,
+      feeCost: market.quoteSplSizeToNumber(event.nativeFeeOrRebate) * (event.eventFlags.maker ? -1 : 1),
+      account: openOrdersAccount,
+      accountSlot: openOrdersSlot,
+      feeTier: feeTier
+    }
+    return fillMessage
+  } else if (event.nativeQuantityPaid.eqn(0)) {
```
Comment (Taylor123) [inline listener/src/serumListener.ts:96]: nit: no need for the else since the other if returns

File: listener/src/serumListener.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      clientId,
+      side,
+      account: openOrdersAccount,
+      accountSlot: openOrdersSlot
+    }
+
+    return doneMessage
+  }
+
+  return
```
Comment (Taylor123) [inline listener/src/serumListener.ts:116]:
```suggestion
```

File: listener/src/serumListener.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    }
+    // map the events to better structure
+    const formattedEvents: EventTypes[] = []
+    const fillsForTradeMatching: Record<string, Fill> = {}
+    events.forEach(event => {
+      const formattedEvent = _mapEventToDataMessage(event, market, timestamp, context.slot)
+      if (formattedEvent) {
+        formattedEvents.push(formattedEvent)
+        /**
+         * Below we capture a Trade event by matching opposite Fill events. This logic
```
Comment (Taylor123) [inline listener/src/serumListener.ts:150]: Is this safe? I'm not familiar enough with the order of events to have an opinion

**Review by Taylor123 — APPROVED** [review]

File: listener/src/graphQLClient.ts (hunk 1 above)
Comment (tomjohn1028) [reply listener/src/graphQLClient.ts:6]: I'm going to keep it like this for now, but I'm thinking it should be made into it's own package for other clients to use

File: listener/src/serumListener.ts (hunk 2 above)
Comment (tomjohn1028) [reply listener/src/serumListener.ts:96]: It's a conditional, so it has purpose

File: listener/src/serumListener.ts (hunk 4 above)
Comment (tomjohn1028) [reply listener/src/serumListener.ts:150]: I'm not sure, but it worked the few times I tested. So I think we roll with it for now and maybe a better pattern will emerge. Plus as long as the _fills_ are in the database you can still figure out the trades

**Pull request merged**